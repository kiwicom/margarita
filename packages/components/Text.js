// @flow

import * as React from 'react';
import { Text as RNText } from 'react-native';
import { StyleSheet } from '@kiwicom/universal-components';

const Text = ({ children }: { children: React.Node }) => {
  return <RNText style={styles.text}>{children}</RNText>;
};

const styles = StyleSheet.create({
  text: {
    alignItems: 'center',
    fontSize: 24,
  },
});

export default Text;
