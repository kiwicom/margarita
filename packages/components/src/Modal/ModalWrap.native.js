// @flow

import * as React from 'react';
import { Modal as RNModal } from 'react-native';

import type { Props } from './Modal';
import ModalContent from './ModalContent';

const ModalWrap = ({ visible, children, onClose }: Props) => {
  return (
    <RNModal
      animationType="slide"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <ModalContent onClose={onClose}>{children}</ModalContent>
    </RNModal>
  );
};

export default ModalWrap;
