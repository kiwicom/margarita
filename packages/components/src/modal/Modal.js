// @flow

import * as React from 'react';
import { View, Platform } from 'react-native';
import {
  StyleSheet,
  Modal as UCModal,
  type StylePropType,
} from '@kiwicom/universal-components';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { Device } from '@kiwicom/margarita-utils';

export type Props = {|
  +children: React.Node,
  +isVisible: boolean,
  +wrapperStyle?: StylePropType,
  +onClose?: () => void,
|};

export default function Modal({
  children,
  isVisible,
  wrapperStyle,
  onClose,
}: Props) {
  const centerContent = Platform.OS === 'web' || Device.isTablet();
  return (
    <UCModal
      isVisible={isVisible}
      onRequestClose={onClose}
      onBackdropPress={onClose}
      style={[styles.modal, centerContent && styles.modalCentered]}
    >
      <View
        style={[
          styles.content,
          centerContent && styles.contentCentered,
          wrapperStyle,
        ]}
      >
        {children}
      </View>
    </UCModal>
  );
}

const styles = StyleSheet.create({
  modal: {
    margin: 0,
    justifyContent: 'flex-end',
  },
  modalCentered: {
    justifyContent: 'center',
  },
  content: {
    width: '100%',
    backgroundColor: defaultTokens.backgroundModal,
    borderTopLeftRadius: parseInt(defaultTokens.borderRadiusLarge, 10),
    borderTopRightRadius: parseInt(defaultTokens.borderRadiusLarge, 10),
    paddingTop: parseInt(defaultTokens.spaceXXSmall, 10),
    paddingBottom: parseInt(defaultTokens.spaceXXLarge, 10),
  },
  contentCentered: {
    maxWidth: 420,
    borderRadius: parseInt(defaultTokens.borderRadiusLarge, 10),
    paddingTop: 0,
    paddingBottom: 0,
  },
});
