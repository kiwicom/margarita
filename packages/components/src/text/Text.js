// @flow

import * as React from 'react';
import { Platform } from 'react-native';
import {
  Text as UniversalComponentsText,
  StyleSheet,
} from '@kiwicom/universal-components';

/**
 * This wrapper is necessary because loading fonts with Expo does not allow using fontWeight and fontStyle
 * cf. https://docs.expo.io/versions/latest/guides/using-custom-fonts/
 */

export default function Text(
  props: $PropertyType<typeof UniversalComponentsText, 'props'>,
) {
  if (Platform.OS !== 'ios' && Platform.OS !== 'android') {
    return <UniversalComponentsText {...props} />;
  }
  const { italic, weight } = props;

  let fontFamily = styles.normal;
  if (italic && weight === 'bold') {
    fontFamily = styles.boldItalic;
  } else if (italic) {
    fontFamily = styles.italic;
  } else if (weight === 'bold') {
    fontFamily = styles.bold;
  }

  return (
    <UniversalComponentsText {...props} style={[props.style, fontFamily]} />
  );
}

const styles = StyleSheet.create({
  normal: {
    fontFamily: 'Roboto',
  },
  bold: {
    fontFamily: 'RobotoBold',
  },
  italic: {
    fontFamily: 'RobotoItalic',
  },
  boldItalic: {
    fontFamily: 'RobotoBoldItalic',
  },
});
