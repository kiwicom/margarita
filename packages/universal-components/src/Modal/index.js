// @flow

export { default as Modal } from './Modal';
