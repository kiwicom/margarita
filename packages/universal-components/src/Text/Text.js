// @flow

import * as React from 'react';
import { Text as RNText, Platform } from 'react-native';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';

import { StyleSheet } from '../PlatformStyleSheet';
import { createStylesGenerator } from '../utils';
import {
  align as alignTypes,
  fontSize,
  fontWeight as fontWeightTypes,
  textColor,
} from './styles';
import type { TextType } from './TextTypes';

const colorGen = createStylesGenerator('color', textColor);
const fontWeightGen = createStylesGenerator('fontWeight', fontWeightTypes);
const fontSizeGen = createStylesGenerator('fontSize', fontSize);
const alignGen = createStylesGenerator('textAlign', alignTypes);

const Text = ({
  align,
  children,
  dataTest,
  italic,
  numberOfLines,
  size,
  style,
  type,
  uppercase,
  weight,
  expo,
}: TextType) => {
  const textStyle = [styles.text];
  if (italic) {
    textStyle.push(styles.italic);
  }
  if (uppercase) {
    textStyle.push(styles.uppercase);
  }
  if (weight) {
    textStyle.push(styles[weight]);
  }
  if (size) {
    textStyle.push(styles[size]);
  }
  if (type) {
    textStyle.push(styles[type]);
  }
  if (align) {
    textStyle.push(styles[align]);
  }
  if (style) {
    textStyle.push(style);
  }

  if (expo && (Platform.OS === 'ios' || Platform.OS === 'android')) {
    let fontFamily = styles.normalFontFamily;
    if (italic && weight === 'bold') {
      fontFamily = styles.boldItalicFontFamily;
    } else if (italic) {
      fontFamily = styles.italicFontFamily;
    } else if (weight === 'bold') {
      fontFamily = styles.boldFontFamily;
    }
    textStyle.push(fontFamily);
  }

  return (
    <RNText
      data-test={dataTest}
      style={textStyle}
      numberOfLines={numberOfLines}
    >
      {children}
    </RNText>
  );
};

Text.defaultProps = {
  expo: false,
};

const styles = StyleSheet.create({
  text: {
    margin: 0,
    fontFamily: Platform.OS === 'web' ? defaultTokens.fontFamily : 'Roboto',
  },
  italic: {
    fontStyle: 'italic',
  },
  uppercase: {
    textTransform: 'uppercase',
  },
  ...fontWeightGen(),
  ...alignGen(),
  ...colorGen(),
  ...fontSizeGen(),
  normalFontFamily: {
    fontFamily: 'Roboto',
  },
  boldFontFamily: {
    fontFamily: 'RobotoBold',
  },
  italicFontFamily: {
    fontFamily: 'RobotoItalic',
  },
  boldItalicFontFamily: {
    fontFamily: 'RobotoBoldItalic',
  },
});

export default Text;
