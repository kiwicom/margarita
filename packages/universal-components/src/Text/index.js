// @flow

export { default as Text } from './Text';
