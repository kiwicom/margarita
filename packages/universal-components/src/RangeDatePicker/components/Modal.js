// @flow

import * as React from 'react';

import { Modal as UCModal } from '../../Modal';
import type { Props } from '../../Modal/ModalTypes';

export default function Modal(props: Props) {
  return <UCModal {...props} />;
}
