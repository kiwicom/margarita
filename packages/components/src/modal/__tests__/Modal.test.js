// @flow

import * as React from 'react';
import { shallow } from 'react-native-testing-library';

import Text from '../../text/Text';
import Modal from '../Modal';

it('renders', () => {
  expect(
    shallow(
      <Modal visible={true} onClose={jest.fn()}>
        <Text>Modal test content</Text>
      </Modal>,
    ),
  ).toMatchSnapshot();
});
