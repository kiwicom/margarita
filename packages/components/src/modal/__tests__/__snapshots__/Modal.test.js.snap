// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <Modal
    backdropOpacity={0.5}
    isVisible={true}
    onBackdropPress={[MockFunction]}
    onRequestClose={[MockFunction]}
    style={
      Array [
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
        },
        false,
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "paddingBottom": 40,
            "paddingTop": 4,
            "width": "100%",
          },
          false,
          undefined,
        ]
      }
    >
      <Text>
        Modal test content
      </Text>
    </Component>
  </Modal>,
}
`;
